report_async/models/report_async.py

```diff
@@ -199,6 +202,22 @@ def run_report(
                 }
             )
         )
+        # save attachment to records
+        if save_attachment_to_records:
+            model = report.model
+            records = self.env[model].browse(docids)
+            for record in records:
+                attachment = self.env["ir.attachment"].sudo().create(
+                    {
+                        "name": out_name,
+                        "datas": out_file,
+                        "type": "binary",
+                        "res_model": model,
+                        "res_id": record.id,
+                    }
+                )
+                if hasattr(record, 'message_post'):
+                    record.message_post(attachment_ids=[attachment.id])
```

Comment:
I'm not 100% sure but I think in some cases this might require a `sudo`, anyway it wouldn't hurt

Reply:
@thomaspaulb added the sudo just in case